src/__tests__/functions.test.ts

```diff
@@ -127,6 +159,7 @@ describe('functions.test.ts', () => {
           .mockImplementation(() => ({})),
         cancelCollectSetupIntent: jest.fn().mockImplementation(() => ({})),
         setSimulatedCard: jest.fn(),
+        collectInputs: jest.fn().mockImplementation(() => ({ collectInputResults: mockCollectInputsReturns })),
```

Comment:
Correct me if I'm understanding this wrong, but this test seems to be mocking `collectInputs` to return `mockCollectInputsReturns`, then it checks that `collectInputs` returns `mockCollectInputsReturns`, which seems to just test that the mocked function is mocked correctly.

What method are we mocking here? I'm a bit confused where `mapFromCollectInputs` from above is being tested.

Reply:
@zakh-stripe, The codes submitted for jest testing is indeed unable to achieve the test effect. We should not use mock functions. We have also tried calling the CollectInputs method in function.ts directly, but it seems that we cannot call the code at the native SDK level. And `mapFromCollectInputs` is not a exposed method which is implemented in native SDKs.

Reply:
Ah, thanks for looking into this, it's concerning that all of the other tests are also doing this same mocking pattern.

Are you able to make a new test file that would have access to the inner `mapFromCollectInputs` method? Maybe you would have to right the test in Swift instead, but that would do the trick

Reply:
Thanks for your guidances. emm..I don't have a workaround with this test yet, without modifying the native part codes. I'll do some research on it. And can we revert this modification of test of this PR, let it just for fixing CollectInputs returns then open a new one for unit test?

Reply:
Okay, just want to double check that you've manually tested the changes and confirmed that the return values are the ones we are putting in our public docs (index.ts)

>And can we revert this modification of test of this PR

I can approve to unblock once this is in, thanks